core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManager.java

```diff
@@ -482,23 +482,26 @@ public boolean doAuthenticate(String userName, Object credential) throws UserSto
                 }
             } else {
                 name = getNameInSpaceForUserName(userName);
-                try {
-                    if (name != null) {
-                        if (debug) {
-                            log.debug("Authenticating with " + name);
+                // if it is the same user DN found in the cache no need of futher authentication required.
+                if (!failedUserDN.equalsIgnoreCase(name)) {
+                    try {
+                        if (name != null) {
+                            if (debug) {
+                                log.debug("Authenticating with " + name);
+                            }
+                            bValue = this.bindAsUser(userName, name, credentialObj);
+                            if (bValue) {
+                                LdapName ldapName = new LdapName(name);
+                                putToUserCache(userName, ldapName);
+                            }
                         }
-                        bValue = this.bindAsUser(userName, name, credentialObj);
-                        if (bValue) {
-                            LdapName ldapName = new LdapName(name);
-                            putToUserCache(userName, ldapName);
+                    } catch (NamingException e) {
+                        String errorMessage = "Cannot bind user : " + userName;
+                        if (log.isDebugEnabled()) {
+                            log.debug(errorMessage, e);
```

Comment:
since the errorMessage is already there, better to add more information to debug log.

Reply:
The change was made to line 486 only to check and skip re-user binding if the same DN has returned from the getNameInSpaceForUserName(userName) method, which is already tried binding using the Cached DN.